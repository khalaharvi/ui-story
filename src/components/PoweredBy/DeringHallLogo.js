import React from 'react';

import { cx } from './PoweredBy';

const DeringHallLogo = () => (
  <svg className={cx('logo')} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 303 31"><title>Dering Hall</title><path fill="#B90A0A" d="M85.6 15.4a9.47 9.47 0 0 0 1.4-5.3A8.449 8.449 0 0 0 83.8 3 11.629 11.629 0 0 0 77 1H64.6a.47.47 0 0 0-.5.5v27.9a.47.47 0 0 0 .5.5h4.9a.47.47 0 0 0 .5-.5V20a.47.47 0 0 1 .5-.5h4.3a.551.551 0 0 1 .5.3l4.9 9.8c.096.187.29.304.5.3h5.5a.517.517 0 0 0 .4-.2.45.45 0 0 0 0-.5l-5.2-10.5a8.337 8.337 0 0 0 4.2-3.3zm-8.8-1.2h-6.4a.999.999 0 0 1-.4-.7V7.4a.47.47 0 0 1 .5-.5h6.4c2.7 0 4.1.9 4.1 3.4 0 3.1-2.3 3.7-4.2 3.9zm-23.4 9.9h-13a.75.75 0 0 1-.5-.7v-5a.47.47 0 0 1 .5-.5h9a.47.47 0 0 0 .5-.5v-3.9a.47.47 0 0 0-.5-.5h-9a.47.47 0 0 1-.5-.5v-5a.63.63 0 0 1 .6-.5h13a.56.56 0 0 0 .5-.6V1.6a.47.47 0 0 0-.5-.5h-19c-.3 0-.5.2-.5.7v27.8c0 .1.2.3.5.3h18.9a.47.47 0 0 0 .5-.5v-4.9a.33.33 0 0 0-.5-.4zm49-23h-4.9a.47.47 0 0 0-.5.5v27.8a.47.47 0 0 0 .5.5h4.9a.47.47 0 0 0 .5-.5V1.6a.47.47 0 0 0-.5-.5zm-92.7 0H.5a.47.47 0 0 0-.5.5v27.8a.47.47 0 0 0 .5.5h8.8a15.67 15.67 0 0 0 9.9-3.2c3.2-2.5 4.9-6.4 4.9-11.2s-1.2-8.4-3.6-10.8-6.1-3.6-10.8-3.6zM16 21.9c-1.5 1.6-3.8 2.4-6.7 2.4H6.5a.47.47 0 0 1-.5-.5V7.2a.47.47 0 0 1 .5-.5h3c6 0 8.6 2.6 8.6 8.7a9.003 9.003 0 0 1-2.1 6.5zM240.2 1.4a.551.551 0 0 0-.5-.3h-4.9a.551.551 0 0 0-.5.3l-10.7 27.8c-.1.2 0 .3.1.5.11.105.25.175.4.2h5.1a.551.551 0 0 0 .5-.3l1.9-5.3a.551.551 0 0 1 .5-.3h10.2a.551.551 0 0 1 .5.3l1.9 5.3c.096.187.29.304.5.3h5.3a.517.517 0 0 0 .4-.2.75.75 0 0 0 .1-.5L240.2 1.4zM233.6 19l2.1-6.3c.5-1.6 1.4-4.2 1.4-4.5 0 .1.9 2.8 1.4 4.5l2.1 6.3h-7zm42.8 5.1h-11a.47.47 0 0 1-.5-.5v-22a.47.47 0 0 0-.5-.5h-4.8a.767.767 0 0 0-.5.2v27.8c0 .6.2.8.5.8h16.8a.47.47 0 0 0 .5-.5v-4.8a.47.47 0 0 0-.5-.5zm-140-23h-4.8a.47.47 0 0 0-.5.5v17.9c-.4-.7-1.5-2.1-2.1-3.1l-.1-.1-10.1-15a.762.762 0 0 0-.4-.2h-4.8a.47.47 0 0 0-.5.5v27.8a.47.47 0 0 0 .5.5h4.8a.47.47 0 0 0 .5-.5V17.2c0-.9.1-2.4.1-3.7 0-1.1-.1-2.1-.1-2.6.4.8 1.2 2.1 1.8 3.1l.1.1 10.5 15.6c.11.105.25.175.4.2h4.7a.47.47 0 0 0 .5-.5V1.6a.54.54 0 0 0-.5-.5zm33 13.9h-9.9a.47.47 0 0 0-.5.5v4a.47.47 0 0 0 .5.5h4.6a6.27 6.27 0 0 1-6 5.1c-3.6 0-7.3-3.2-7.3-9.2 0-5.5 3-9.4 7.3-9.4 2.7 0 4.6 1.2 6.1 3.9.1.1.2.2.3.2a.597.597 0 0 0 .4-.1l4.1-2.4a.571.571 0 0 0 .2-.7C166.7 3 163 .8 158.2.8a12.63 12.63 0 0 0-9.3 4c-2.5 2.7-3.8 6.5-3.8 11.1s1.3 8.3 3.8 10.9a12.531 12.531 0 0 0 9.2 3.9 11.623 11.623 0 0 0 8.5-3.5c2.5-2.7 3.6-6.5 3.3-11.7a.54.54 0 0 0-.5-.5zm45.2-13.9h-5a.47.47 0 0 0-.5.5v12a.47.47 0 0 1-.5.5h-10.1a.47.47 0 0 1-.5-.5v-12a.47.47 0 0 0-.5-.5h-4.8a.56.56 0 0 0-.6.5v27.8a.56.56 0 0 0 .6.5h4.8a.47.47 0 0 0 .5-.5v-9a.47.47 0 0 1 .5-.5h10.1a.47.47 0 0 1 .5.5v9a.47.47 0 0 0 .5.5h5a.47.47 0 0 0 .5-.5V1.6a.54.54 0 0 0-.5-.5zm87.8 23h-11a.47.47 0 0 1-.5-.5v-22a.47.47 0 0 0-.5-.5h-4.8a.767.767 0 0 0-.5.2v27.8c0 .6.2.8.5.8h16.8a.47.47 0 0 0 .5-.5v-4.8a.47.47 0 0 0-.5-.5z" /></svg>
);

export default DeringHallLogo;
