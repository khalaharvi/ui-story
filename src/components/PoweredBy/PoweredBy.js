import React from 'react';
import PropTypes from 'prop-types';

import { createClassNames } from 'utils';
import Link from 'components/Link';
import DeringHallLogo from './DeringHallLogo';

export const cx = createClassNames('poweredby');

const PoweredBy = ({ to }) => (
  <div className={cx('')}>
    <Link className={cx('link')} to={to}>
      <small>Powered By</small>
      <DeringHallLogo />
    </Link>
  </div>
);

PoweredBy.propTypes = {
  to: PropTypes.string
};

PoweredBy.defaultProps = {
  to: 'https://deringhall.com?ref=ui-kit'
};

export default PoweredBy;
