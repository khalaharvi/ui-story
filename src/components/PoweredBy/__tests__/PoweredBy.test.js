import React from 'react';
import { shallow } from 'enzyme';

import PoweredBy from '../PoweredBy';

describe('<PoweredBy />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PoweredBy />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with custom to prop', () => {
    const wrapper = shallow(<PoweredBy to="https://encrypted.google.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
