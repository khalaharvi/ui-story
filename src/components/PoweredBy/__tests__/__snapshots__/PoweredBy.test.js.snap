// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PoweredBy /> renders correctly 1`] = `
<div
  className="dh-poweredby"
>
  <Link
    className="dh-poweredby-link"
    external={false}
    noFollow={false}
    to="https://deringhall.com?ref=ui-kit"
  >
    <small>
      Powered By
    </small>
    <DeringHallLogo />
  </Link>
</div>
`;

exports[`<PoweredBy /> renders correctly with custom to prop 1`] = `
<div
  className="dh-poweredby"
>
  <Link
    className="dh-poweredby-link"
    external={false}
    noFollow={false}
    to="https://encrypted.google.com"
  >
    <small>
      Powered By
    </small>
    <DeringHallLogo />
  </Link>
</div>
`;
